SharedPackages/BrowserServicesKit/Sources/Subscription/Storage/V2/KeychainManager.swift

```diff
@@ -261,63 +259,59 @@ public final class KeychainManager: KeychainManaging {
+            Logger.keychainManager.error("Failed to update keychain item: \(status.humanReadableDescription, privacy: .public), adding data to writing queue")
             addToWritingBacklog(data, forKey: key)
         default:
             removeFromWritingBacklog(forKey: key)
-            Logger.keychainManager.error("SecItemUpdate failed with status: \(status.humanReadableDescription) for field: \(key)")
+            Logger.keychainManager.error("SecItemUpdate failed with status: \(status.humanReadableDescription, privacy: .public) for field: \(key)")
             throw AccountKeychainAccessError.keychainSaveFailure(status)
         }
     }
 
     // MARK: - Notification Handling
 
-    /// Sets up platform-specific notifications to detect when the keychain becomes available.
-    /// 
+    /// Sets up platform-specific notifications to detect when the keychain becomes available, when the app goes in background or is terminated.
     /// This enables automatic retry of operations that were queued when the keychain was unavailable.
     private func setupKeychainAvailabilityNotifications() {
-        #if canImport(UIKit)
-        // On iOS, listen for app becoming active and protected data becoming available
-        NotificationCenter.default.publisher(for: UIApplication.didBecomeActiveNotification)
-            .receive(on: accessQueue)
-            .sink { [weak self] _ in
-                self?.processWritingBacklog()
-            }
-            .store(in: &cancellables)
-
-        NotificationCenter.default.publisher(for: UIApplication.protectedDataDidBecomeAvailableNotification)
-            .receive(on: accessQueue)
-            .sink { [weak self] _ in
-                self?.processWritingBacklog()
-            }
-            .store(in: &cancellables)
-
-        Logger.keychainManager.debug("Set up iOS keychain availability notifications")
-
-        #elseif canImport(AppKit)
-        // On macOS, listen for app becoming active and workspace session becoming active
-        NotificationCenter.default.publisher(for: NSApplication.didBecomeActiveNotification)
-            .receive(on: accessQueue)
-            .sink { [weak self] _ in
-                self?.processWritingBacklog()
-            }
-            .store(in: &cancellables)
+        Logger.keychainManager.log("Set up keychain availability and recovery notifications")
 
-        NotificationCenter.default.publisher(for: NSWorkspace.sessionDidBecomeActiveNotification)
-            .receive(on: accessQueue)
-            .sink { [weak self] _ in
-                self?.processWritingBacklog()
-            }
-            .store(in: &cancellables)
-
-        Logger.keychainManager.debug("Set up macOS keychain availability notifications")
-
-        #else
-        Logger.keychainManager.info("Keychain notifications not supported on this platform")
-        #endif
+#if canImport(UIKit)
+        Publishers.MergeMany(
+            // Foreground notifications
```

Comment:
Here, I just grouped the notifications and added some more as a best effort attempt to attempt to write the backlog in the keychain as often as possible